nemo/core/optim/lr_scheduler.py

```diff
@@ -681,18 +681,19 @@ def prepare_lr_scheduler(
 
             # Remove extra parameters from scheduler_args nest
             # Assume all other parameters are to be passed into scheduler constructor
-
-            if 'name' in scheduler_args and scheduler_args['name'] == 'ReduceLROnPlateau':
-                add_max_args_flag = False
-                interval = 'epoch'
-
             scheduler_args.pop('name', None)
             scheduler_args.pop('t_max_epochs', None)
             scheduler_args.pop('t_accumulate_grad_batches', None)
             scheduler_args.pop('t_limit_train_batches', None)
             scheduler_args.pop('t_num_workers', None)
             scheduler_args.pop('monitor', None)
             scheduler_args.pop('reduce_on_plateau', None)
+        
```

Comment:
Avoid this and just do use python inspect module to check if the scheduler needs max_steps or not inside of get scheduler. Much simpler logic.

Reply:
Thanks for the suggestion, I have changed to using `inspect.signature()` instead. By the way, why do we pass the `scheduler_args` twice, in `scheduler_cls = get_scheduler(scheduler_name, **scheduler_args)` and `schedule = scheduler_cls(optimizer, **scheduler_args)`?

Reply:
One is to get a class, another is to instantiate the class with those kwargs